tasks/main.yml

```diff
@@ -26,17 +26,23 @@
       - docker
     state: latest
 
+- name: Test to see if Docker group is already installed
+  shell: grep "^docker" /etc/group
+  register: test_docker_grep
+
 - name: Docker setup-groupadd
   command: groupadd docker
   become: true
   become_user: root
   become_method: sudo
+  when: test_docker_grep.stdout == ""
 
 - name: Docker setup-jenkins user
   command: usermod -a -G docker jenkins
   become: true
   become_user: root
   become_method: sudo
+  when: test_docker_grep.stdout == ""
```

Comment:
Have you looked into using [the `group` option of the `user` module](https://docs.ansible.com/ansible/latest/user_module.html#options) to handle all of this?

Reply:
Anyway, we can handle that as a follow-up.